### PR TITLE
feat: implements AppMemoryUsage and AppMemoryUsageLimit for Android

### LINKED ISSUE #5891: Please add MemoryManager
## What would you like to be added:
MemoryManager, at least .AppMemoryUsage and .AppMemoryUsageLimit

## Why is this needed:
I'm using it in my app (and I'm porting app to Uno).

## For which Platform:

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Hudney**
GitHub Issue: closes #5891

## PR Type
- Feature

## What is the current behavior?
Not implemented, throws an exception. #5891

## What is the new behavior?
Returns the correct values.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Uno.csproj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="MSBuild.Sdk.Extras" ToolsVersion="15.0">
+﻿<Project Sdk="MSBuild.Sdk.Extras" ToolsVersion="15.0">
```
Comment (jeromelaban) [inline src/Uno.UWP/Uno.csproj:1]: You can rollback all the modifications made to this file. Visual Studio makes them, but they are not useful.

File: src/Uno.UWP/System/MemoryManager.Android.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+	partial class MemoryManager
+	{
+		public static ulong AppMemoryUsage
+		{
+			get
+			{
+				Debug.MemoryInfo mi = new Debug.MemoryInfo();
+				Debug.GetMemoryInfo(mi);
+				var totalMemory = mi.NativePrivateDirty + mi.OtherSharedDirty;
```
Comment (pkar70) [inline src/Uno.UWP/System/MemoryManager.Android.cs:16]: a) Why not add also dalvikPrivateDirty?
b) OtherSharedDirty is, according to doc, in 1 KiB pages, not in bytes. So, multiply it by 1024
c) data from Debug.MemoryInfo are for process. Does Uno Android app has only one process? Or should we iterate processes per app?

Comment (teras) [reply src/Uno.UWP/System/MemoryManager.Android.cs:16]: a+b) I think the correct way to do it is using totalPss:  https://developer.android.com/reference/android/os/Debug.MemoryInfo#getTotalPss() and multiply by 1024. To have an exact memory consumption for a Unix process (taking into consideration shared memory) is something there's no direct answer to it.

c) There's a concept of multi-process applications in Android, but you need specific code to support this and involves multiple & "remote" activities. Maybe @jeromelaban is more proficient on this, but I doubt that Uno has even remote interest on supporting any of this. And if I remember correctly after Android M (?) the OS provides you with a single PID when you ask for your application PIDs. So, IMHO I think this method method is the best bet to provide the information of the memory consumption of an Application.

Comment (jeromelaban) [reply src/Uno.UWP/System/MemoryManager.Android.cs:16]: That's correct, multi-process is not of interest at this point, particularly when matching this particular WinRT API. If an application needs fined grained information, it'll need to go through the native api directly.

**Comment by jeromelaban** [comment]
Merged in https://github.com/unoplatform/uno/pull/7048

**Pull request closed without merging**